internal/provider/user_resource.go

```diff
@@ -0,0 +1,371 @@
+		State             types.String `tfsdk:"state"`
+		Email             types.String `tfsdk:"email"`
+		AccountAccess     types.String `tfsdk:"account_access"`
+		NamespaceAccesses types.List   `tfsdk:"namespace_accesses"`
+
+		Timeouts timeouts.Value `tfsdk:"timeouts"`
+	}
+
+	userNamespaceAccessModel struct {
+		Namespace  types.String `tfsdk:"namespace"`
+		Permission types.String `tfsdk:"permission"`
+	}
+)
+
+var (
+	_ resource.Resource                = (*userResource)(nil)
+	_ resource.ResourceWithConfigure   = (*userResource)(nil)
+	_ resource.ResourceWithImportState = (*userResource)(nil)
+
+	userNamespaceAccessAttrs = map[string]attr.Type{
+		"namespace":  types.StringType,
+		"permission": types.StringType,
+	}
+)
+
+func NewUserResource() resource.Resource {
+	return &userResource{}
+}
+
+func (r *userResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(cloudservicev1.CloudServiceClient)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected cloudservicev1.CloudServiceClient, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+func (r *userResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_user"
+}
+
+func (r *userResource) Schema(ctx context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Provisions a Temporal Cloud user.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The unique identifier of the user.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"state": schema.StringAttribute{
+				Description: "The current state of the user.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"email": schema.StringAttribute{
+				Description: "The email address for the user.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"account_access": schema.StringAttribute{
+				Description: "The role on the account. Must be one of [admin, developer, read] (case-insensitive)",
+				Required:    true,
+				Validators: []validator.String{
+					stringvalidator.OneOfCaseInsensitive("admin", "developer", "read"),
+				},
+			},
+			"namespace_accesses": schema.ListNestedAttribute{
+				Description: "The list of namespace accesses.",
+				Optional:    true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"namespace": schema.StringAttribute{
```

Comment:
should this be the `name` property of the namespace or the `id` property? Based on the example it looks like it's `id`, so that might be worth calling out explicitly here ("The ID of the namespace to assign permissions to")

Reply:
Definitely should be the ID. I'll rename the attribute to `namespace_id` so it is obvious.